pages/chaoss/visualizations/contrib_importance_pie.py

```diff
@@ -0,0 +1,248 @@
+                                            
+                                            )
+                                    ], 
+                                    className = "me-2",
+                                ),
+                                 dbc.Col(
+                                    [dbc.Button(
+                                        "About Graph",
+                                        id=f"popover-target-{PAGE}-{VIZ_ID}",
+                                        color="secondary",
+                                        size="sm",
+                                    ), ],
+                                    width="auto",
+                                    style={"paddingTop": ".5em"},
+                                ),
+                            ], 
+                            align="center",
+                        ), 
+                    ]
+                ),
+            ]
+        )
+    ],
+)
+
+# callback for graph info popover
+@callback(
+    Output(f"popover-{PAGE}-{VIZ_ID}", "is_open"),
+    [Input(f"popover-target-{PAGE}-{VIZ_ID}", "n_clicks")],
+    [State(f"popover-{PAGE}-{VIZ_ID}", "is_open")],
+)
+def toggle_popover(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+# callback for contrib-importance graph
+@callback(
+    Output(f"{PAGE}-{VIZ_ID}", "figure"),
+    Output(f"check-alert-{PAGE}-{VIZ_ID}", "is_open"),
+    [
+        Input("repo-choices", "data"),
+        Input(f"action-type-{PAGE}-{VIZ_ID}", "value"),
+        Input(f"patterns-{PAGE}-{VIZ_ID}", "value"),
+        Input(f"top-k-contributors-{PAGE}-{VIZ_ID}", "value"),
+
+    ],
+    background=True,
+)
+
+def create_top_10_cntrbs_graph(repolist, action_type, patterns, top_k):
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=ctq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=ctq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.warning(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.warning(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph, False
+    
+    # checks if there is a contribution of a specfic action type in repo set
+    if not df["Action"].str.contains(action_type).any():
+        return dash.no_update, True
+    
+    # function for all data pre processing
+    df = process_data(df, action_type, patterns, top_k)
+
+    fig = create_figure(df, action_type)
+
+    logging.warning(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig, False
+
+def process_data(df: pd.DataFrame, action_type, patterns, top_k):
+    # subset the df such that it only contains rows where the column value for Action is the action type
+    df = df[df["Action"].str.contains(action_type)]
+
+    # remove substring matches between login and patterns
+    df = df[~df['login'].isin(patterns)]
```

Comment:
Should this be in the same format as line 200? I thought isin would require that df.login (in this example) string is in the patterns list, no the subset

Reply:
Double checked and isin checks if df.login is an exact match and NOT substring matches to any element in the patterns list. Edited my comment  in line 208: `# remove rows where the login column value is in patterns - requires the login string to be in patterns`

Reply:
So this should be changed to .str.contains(patterns) as we are looking for substrings and that is what the search bar is prompting for